python/lsst/pipe/tasks/makeCoaddTempExp.py

```diff
@@ -538,3 +553,194 @@ def applySkyCorr(self, dataRef, calexp):
+        outputDataIds : `dict`
+            Keys are the names of the configs describing input dataset types.
+            Values are DataIds (or lists of DataIds) that task is to produce
+            for corresponding dataset type.
+        butler : `lsst.daf.butler.Butler`
+            Gen3 Butler object for fetching additional data products before
+            running the Task
+
+        Returns
+        -------
+        result : `lsst.pipe.base.Struct`
+           Result struct with components:
+
+           - ``direct`` : (optional) direct Warp Exposure
+                          (``lsst.afw.image.Exposure``)
+           - ``psfMatched``: (optional) PSF-Matched Warp Exposure
+                            (``lsst.afw.image.Exposure``)
+        """
+        # Construct skyInfo expected by `run`
+        skyMap = inputData["skyMap"]
+        outputDataId = next(iter(outputDataIds.values()))
+        inputData['skyInfo'] = makeSkyInfo(skyMap,
+                                           tractId=outputDataId['tract'],
+                                           patchId=outputDataId['patch'])
+
+        # Construct list of DataId's expected by `run`
+        dataIdList = inputDataIds['calExpList']
+        inputData['dataIdList'] = dataIdList
+
+        # Construct list of ccdIds expected by `run`
+        inputData['ccdIdList'] = [dataId['detector'] for dataId in dataIdList]
+
+        # Extract integer visitId requested by `run`
+        visits = [dataId['visit'] for dataId in dataIdList]
+        assert(all(visits[0] == visit for visit in visits))
+        inputData["visitId"] = visits[0]
+
+        self.prepareCalibratedExposures(**inputData)
+        results = self.run(**inputData)
+        return pipeBase.Struct(**results.exposures)
+
+    def prepareCalibratedExposures(self, calExpList, backgroundList=None, skyCorrList=None, **kwargs):
+        """Calibrate and add backgrounds to input calExpList in place
+
+        TODO DM-17062: apply jointcal/meas_mosaic here
+
+        Parameters
+        ----------
+        calExpList : `list` of `lsst.afw.image.Exposure`
+            Sequence of calexps to be modified in place
+        backgroundList : `list` of `lsst.afw.math.backgroundList`
+            Sequence of backgrounds to be added back in if bgSubtracted=False
+        skyCorrList : `list` of `lsst.afw.math.backgroundList`
+            Sequence of background corrections to be subtracted if doApplySkyCorr=True
+        """
+        backgroundList = len(calExpList)*[None] if backgroundList is None else backgroundList
+        skyCorrList = len(calExpList)*[None] if skyCorrList is None else skyCorrList
+        for calexp, background, skyCorr in zip(calExpList, backgroundList, skyCorrList):
+            mi = calexp.maskedImage
+            if not self.config.bgSubtracted:
+                mi += background.getImage()
+            if self.config.doApplySkyCorr:
+                mi -= skyCorr.getImage()
+            del mi
```

Comment:
I assume it's just copied from somewhere else, but I'm pretty confident this line is totally unnecessary; reassigning the variable or ending the function will also make it available for garbage collection.